packages/react-search-ui-views/src/view-helpers/getUrlSanitizer.js

```diff
@@ -0,0 +1,33 @@
+const VALID_PROTOCOLS = ["http:", "https:"];
+
+/**
+ *
+ * @param {URL} URLParser URL interface provided by browser https://developer.mozilla.org/en-US/docs/Web/API/URL
+ * @param {String} currentLocation String representation of the browser's current location
+ */
+export default function getUrlSanitizer(URLParser, currentLocation) {
+  // This function is curried so that dependencies can be injected and don't need to be mocked in tests.
+  return url => {
+    if (!url) return url;
```

Comment:
This check isn't strictly needed because if the parsing later on fails, we'll return an empty string (which isn't exactly the same as this).

Reply:
I put this in because if this is not here, it attempts to parse an empty string as a relative url, so it ends up returning `http://example.com/undefined` or `http://example.com/`, etc

Reply:
Oh right, I didn't consider the relative case :v:

Reply:
Just saw https://github.com/elastic/search-ui/pull/481#discussion_r408185113 and yes, that change would also make this one possible.

Reply:
https://github.com/elastic/search-ui/pull/481/commits/57655049d66b550aa135add7d8088571d37f4442